### PR TITLE
Queue multi action

### PR DESCRIPTION
**Pull request by snichme**
Continuation on the work that Christina did before starting the thesis. 

Popup is shown when a checkbox is selected, one request per queue

TODO: 
- Checkbox in header that checks all queues

image

### CODE REVIEW

**Comment by snichme** [comment]
It's a very big diff since I've moved the js into it's own file but the changes are on this lines: https://github.com/cloudamqp/avalanchemq/pull/330/files#diff-61e046fd549fa0d149192cf50d76a706bb66e85185b34a59760be34d5e644875R49-R68

File: static/js/queues.js [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  }
+  const performMultiAction = (el) => {
+    const action = el.target.dataset.action
+    const elems = document.querySelectorAll("input[data-name]:checked")
+    const totalCount = elems.length
+    let performed = 0
+    elems.forEach(el => {
+      const data = el.dataset;
+      const urlEncodedQueue = encodeURIComponent(data.name)
+      const urlEncodedVhost = encodeURIComponent(data.vhost)
```
Comment (jage) [inline static/js/queues.js:24]: Data elements already encoded, we double-encode now (breaking vhost `/`).

**Review by jage — COMMENTED** [review]
Nice!

* Inline comment about double-encoded name/vhost
* If I select all, then purge, the "select all" checkbox is still selected
* Would be nice to get a "Close" or "X" link to close the action window if you change your mind

**Comment by jage** [comment]
Not prioritised, but couldn't help myself to google this https://stackoverflow.com/questions/659508/how-can-i-shift-select-multiple-checkboxes-like-gmail :-)

**Comment by snichme** [comment]
Was trying out the shift + click but felt that it's better to get this out and then add the shift select later (keep it simpler). 

But I know that people want it and it is a very nice feature.

**Comment by jage** [comment]
> Was trying out the shift + click but felt that it's better to get this out and then add the shift select later (keep it simpler).

100% agree :)

File: src/avalanchemq/queue/queue.cr [hunk 2]
```diff
@@ -893,6 +893,7 @@ module AvalancheMQ
         end
         builder.field("consumer_details") do
           builder.array do
+            # pp @consumers.each
```
Comment (jage) [inline src/avalanchemq/queue/queue.cr:896]: Reminder to remove

File: static/queues.html [hunk 3]
```diff
@@ -24,11 +24,19 @@ <h2>
     <aside></aside>
     <main>
       <section class="card">
+        <div id="multiselect-controls" class="popup-card">
```
Comment (jage) [inline static/queues.html:27]:
```suggestion
        <div id="multiselect-controls" class="popup-card hide">
```

Accidentally removed during extraction to CSS class.

**Review by jage — APPROVED** [review]

**Pull request merged**